src/centerline/geometry.py

```diff
@@ -82,14 +83,15 @@ def _construct_centerline(self):
                     x=vertices[ridge[1]][0], y=vertices[ridge[1]][1]
                 )
                 linestring = LineString((starting_point, ending_point))
+                linestrings.append(linestring)
 
-                if self._linestring_is_within_input_geometry(linestring):
-                    linestrings.append(linestring)
-
-        if len(linestrings) < 2:
+        str_tree = STRtree(linestrings)
+        linestrings_indexes = str_tree.query(self._input_geometry, 'contains')
+        contained_linestrings = [linestrings[i] for i in linestring_indexes]
```

Comment:
You have a typo here. The variable is `linestrings_indexes`, but `linestring_indexes` is used.

Reply:
Yeah sorry about that - I tried to make the linting fix on the github browser instead of in an ide.

I believe it should be fixed now.

Reply:
No worries. Could you please run `black .` from the repo's root, commit and push?

Reply:
Done - 

I had tried to install tox on my machine to run everything but I wasn't able to get it up. Sorry about that. Ran black now.